scylla/src/response/rows_displayer.rs

```diff
@@ -0,0 +1,1194 @@
+            };
+            let columns: Vec<Option<CqlValue>> = row_result.columns;
+            let mut row_values: Vec<Box<dyn StringConvertible>> = Vec::new();
+            let columns = columns.iter().map(|item| {
+                let item = match item {
+                    Some(item) => item,
+                    None => &CqlValue::Empty,
+                };
+                get_item_wrapper(item, &self.display_settings)
+            });
+            row_values.extend(columns);
+
+            builder.push_record(row_values);
+        }
+
+        // build the table
+        let mut binding = builder.build();
+        let mut table = binding.with(Style::psql()).with(Alignment::right());
+        if self.display_settings.terminal_width != 0 {
+            table = table.with(
+                Width::wrap(self.display_settings.terminal_width).priority(Priority::max(true)),
+            );
+        }
+
+        // align and colorize the table
+        table.modify(Rows::first(), Alignment::left());
+
+        if self.display_settings.print_in_color {
+            table = table.with(Colorization::exact(
+                [Color::FG_MAGENTA | Color::BOLD],
+                tabled::settings::object::Rows::first(),
+            ));
+        }
+
+        write!(f, "{}", table)
+    }
+}
+
+/// Settings for displaying rows in a `RowsDisplayer`.
+///
+/// This struct holds various configuration options for formatting and displaying
+/// rows received from the database. It includes settings for byte display format,
+/// exponent display for floats and integers, precision for doubles, color usage,
+/// and terminal width for wrapping the table output.
+struct RowsDisplayerSettings {
+    byte_displaying: ByteDisplaying,    // for blobs
+    exponent_displaying_floats: bool,   // for floats
+    exponent_displaying_integers: bool, // for integers
+    double_precision: usize,            // for doubles
+    print_in_color: bool,
+    terminal_width: usize,
+    colors: ValueColors,
+}
+
+impl RowsDisplayerSettings {
+    fn new() -> Self {
+        Self {
+            byte_displaying: ByteDisplaying::Hex,
+            exponent_displaying_floats: false,
+            exponent_displaying_integers: false,
+            double_precision: 5,
+            print_in_color: true,
+            terminal_width: 0,
+            colors: ValueColors::default(),
+        }
+    }
+}
+
+impl Default for RowsDisplayerSettings {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[derive(Clone, Debug)]
+struct ValueColors {
+    pub text: &'static str,
+    pub error: &'static str,
+    pub blob: &'static str,
+    pub timestamp: &'static str,
+    pub date: &'static str,
+    pub time: &'static str,
+    pub int: &'static str,
+    pub float: &'static str,
+    pub decimal: &'static str,
+    pub inet: &'static str,
+    pub boolean: &'static str,
+    pub uuid: &'static str,
+    pub duration: &'static str,
+    pub collection: &'static str,
+}
```

Comment:
:wrench: As the struct itself is mod-private, then its fields should not be more public than the struct itself. Please remove the `pub`s.